tpl/collections/collections.go

```diff
@@ -533,31 +533,55 @@ func (ns *Namespace) Slice(args ...any) any {
 
 type dKey struct {
 	seed uint64
-	n    int
-	hi   int
+	n    uint64
+	hi   uint64
 }
 
-// D returns a slice of n unique random numbers in the range [0, hi) using the provded seed,
-// using  J. S. Vitter's Method D for sequential random sampling, from Vitter, J.S.
-// - An Efficient Algorithm for Sequential Random Sampling - ACM Trans. Math. Software 11 (1985), 37-57.
-// See  https://getkerf.wordpress.com/2016/03/30/the-best-algorithm-no-one-knows-about/
-func (ns *Namespace) D(seed, n, hi int) []int {
-	key := dKey{seed: cast.ToUint64(seed), n: n, hi: hi}
-	if key.n <= 0 || key.hi <= 0 || key.n > key.hi {
-		return nil
+// D returns a slice of unique random integers, sorted from low to high, in the
+// half-open interval [0, hi) using the provided seed value. The number of
+// elements in the resulting slice is n or hi, whichever is less.
+//
+// If n <= hi, this function returns a sorted random sample of size n using
+// J. S. Vitter’s Method D for sequential random sampling.
+//
+// If n > hi, this function returns the full, sorted range [0, hi) of size hi.
+//
+// Reference:
+//
+//	J. S. Vitter, "An efficient algorithm for sequential random sampling," ACM Trans. Math. Softw., vol. 11, no. 1, pp. 37–57, 1985.
+//	See also: https://getkerf.wordpress.com/2016/03/30/the-best-algorithm-no-one-knows-about/
+func (ns *Namespace) D(seed, n, hi uint64) ([]int, error) {
+	if n <= 0 {
+		return nil, fmt.Errorf("the number of requested values (%d) must be greater than zero", n)
 	}
-	if key.n > maxSeqSize {
-		panic(errSeqSizeExceedsLimit)
+	if hi <= 0 {
+		return nil, fmt.Errorf("the maximum requested value (%d) must be greater than zero", hi)
```

Comment:
I don't think we want to make 0 an error for `n` or `hi`. If that was discussed in the original issue, I must have misread it ... Ref the "common use case" in the docs, I think it's not uncommon (or at least not an error case) to run that on some empty collection. I would expect both of these to return an empty slice.

```handlebars
  {{ $p := site.RegularPages }}
  {{ range collections.D time.Now.YearDay 5 ($p | len) }}
    {{ with (index $p .) }}
      <li><a href="{{ .RelPermalink }}">{{ .LinkTitle }}</a></li>
    {{ end }}
  {{ end }}
```